vocode/streaming/transcriber/deepgram_transcriber.py

```diff
@@ -24,48 +25,66 @@
 )
 from vocode.streaming.utils.worker import AsyncWorker
 from websockets.client import WebSocketClientProtocol
+from numpy.typing import DTypeLike
+from pydantic import BaseModel
 
 PUNCTUATION_TERMINATORS = [".", "!", "?"]
 MAX_SILENCE_DURATION = 2.0
 NUM_RESTARTS = 5
 
 # Silero VAD setup
+pp = PrettyPrinter(indent=2, width=80)
 model, utils = torch.hub.load(repo_or_dir="snakers4/silero-vad", model="silero_vad")
 (get_speech_timestamps, save_audio, read_audio, VADIterator, collect_chunks) = utils
 
-# Constants for Silero VAD
-USE_INT16 = False
-SAMPLE_RATE = 16000 if USE_INT16 else 8000
-DTYPE = np.int16 if USE_INT16 else np.int8
-CHUNK = 512 if USE_INT16 else 256
-WINDOWS = 3
-WINDOW_SIZE = WINDOWS * SAMPLE_RATE // CHUNK
-PREFIX_PADDING_MS = 150  # Minimum speech duration to trigger detection
-GRACE_PERIOD_MS = 300  # Grace period before resetting speech detection
+
+class AudioEncodingModel(BaseModel):
+    name: str
+    chunk: int
+    dtype: DTypeLike
+    vad_sampling_rate: int
+
+    class Config:
+        arbitrary_types_allowed = True
 
 
 class VADWorker(AsyncWorker):
+
     def __init__(
         self,
         input_queue: asyncio.Queue,
         output_queue: asyncio.Queue,
-        transcriber: BaseAsyncTranscriber,
+        transcriber: "DeepgramTranscriber",
```

Comment:
It is a string because that how you define types, which may be a circular reference, in your editor it displays it properly as a type. It is the same as the type
![image](https://github.com/user-attachments/assets/6472dc6b-3e75-4d02-9fd8-9aacbac7f24d)